### PR TITLE
DOC: Fix title capitalization in documentation files (#32550)

### PR DESCRIPTION
**Pull request by Wong2333**
Corrected title capitalization in various .rst files to match the standard of capitalizing only the first word

- doc\source\whatsnew\v0.12.0.rst
- doc\source\whatsnew\v0.25.0.rst
- doc\source\whatsnew\v1.2.0.rst
- doc\source\whatsnew\v3.0.0.rst

Fixes part of issue https://github.com/pandas-dev/pandas/issues/32550.

### CODE REVIEW

**Comment by Wong2333** [comment]
Fix title capitalization in documentation files

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -551,7 +551,7 @@ with :attr:`numpy.nan` in the case of an empty :class:`DataFrame` (:issue:`26397
 ``__str__`` methods now call ``__repr__`` rather than vice versa
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-pandas has until now mostly defined string representations in a pandas objects'
+Pandas has until now mostly defined string representations in a pandas objects'
```
Comment (mroeschke) [inline doc/source/whatsnew/v0.25.0.rst:554]:
```suggestion
pandas has until now mostly defined string representations in a pandas objects'
```

pandas is supposed to be lowercase (and throughout this file)

Comment (Wong2333) [reply doc/source/whatsnew/v0.25.0.rst]: Ok, I've corrected it

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Wong2333

**Pull request merged**